examples/tv_minimization_reconstruction_pdhg.py

```diff
@@ -0,0 +1,190 @@
+#
+# In the following, we load some 2D radial MR data and use the just described identification to set up
+# the corresponding problem to be solved with PDHG.
+
+# %%
+import mrpro
+
+# %% [markdown]
+# ##### Read-in the raw data
+
+# %%
+# Use the trajectory that is stored in the ISMRMRD file
+trajectory = mrpro.data.traj_calculators.KTrajectoryIsmrmrd()
+# Load in the Data from the ISMRMRD file
+kdata = mrpro.data.KData.from_file(data_file.name, trajectory)
+kdata.header.recon_matrix.x = 256
+kdata.header.recon_matrix.y = 256
+
+# %% [markdown]
+# ### Set up the operator $A$
+# Estimate coil sensitivity maps and density compensation function. Also run a direct (adjoint)
+# reconstruction and iterative SENSE as methods of comparison.
+
+# %%
+# Set up direct reconstruction class to estimate coil sensitivity maps, density compensation etc
+direct_reconstruction = mrpro.algorithms.reconstruction.DirectReconstruction(kdata)
+img_direct = direct_reconstruction(kdata).data
+
+# run tierative SENSE to get an initiual guess of the solution
+iterative_sense_reconstruction = mrpro.algorithms.reconstruction.IterativeSENSEReconstruction(
+    kdata, n_iterations=8, csm=direct_reconstruction.csm
+)
+img_iterative_sense = iterative_sense_reconstruction(kdata).data
+
+# Define Fourier operator using the trajectory and header information in kdata
+fourier_operator = mrpro.operators.FourierOp.from_kdata(kdata)
+
+# estimate density compensation
+dcf_operator = mrpro.data.DcfData.from_traj_voronoi(kdata.traj).as_operator()
+
+# Calculate coil maps
+img_coilwise = mrpro.data.IData.from_tensor_and_kheader(*fourier_operator.H(*dcf_operator(kdata.data)), kdata.header)
+csm_operator = mrpro.data.CsmData.from_idata_walsh(img_coilwise).as_operator()
+
+# Create the entire acquisition operator A
+acquisition_operator = fourier_operator @ csm_operator
+
+# adjoint and iterative SENSE reconstruction as comparison methods
+# adjoint reconstruction
+(img_adjoint,) = acquisition_operator.H(kdata.data)
+
+
+# %% [markdown]
+# ### Recast the problem to be able to apply PDHG
+
+# %%
+# Define the gradient operator
+from mrpro.operators import FiniteDifferenceOp
+
+# The operator computes the directional derivatives along the penultimate and last dimensions (x,y)
+nabla_operator = FiniteDifferenceOp(dim=(-2, -1), mode='forward')
+
+# Set up the problem by using the previously described identification
+from mrpro.algorithms.optimizers import pdhg
+from mrpro.operators import LinearOperatorMatrix, ProximableFunctionalSeparableSum
+from mrpro.operators.functionals import L1NormViewAsReal, L2NormSquared, ZeroFunctional
+
+# Regularization parameter for the $\ell_1$-norm
+regularization_parameter = 0.5
+
+# Define the functionals
+kdata_tensor = kdata.data
+l2 = 0.5 * L2NormSquared(target=kdata_tensor, divide_by_n=True)
+l1 = regularization_parameter * L1NormViewAsReal(divide_by_n=True)
+
+# Define the functionals f and g and the operator K
```

Comment:
Either make it headins or consider more helpful comments, describing why you do something. Not what.
it is quite obvious that you define f,g, and operator.